gtsfm/utils/io.py

```diff
@@ -182,10 +200,40 @@ def write_images(gtsfm_data: GtsfmData, save_dir: str) -> None:
             camera = gtsfm_data.get_camera(i)
             wRi_quaternion = camera.pose().rotation().quaternion()
             wti = camera.pose().translation()
+
             tx, ty, tz = wti
             qw, qx, qy, qz = wRi_quaternion
+            gtsfm_rot = R.from_quat([qx, qy, qz, qw]).as_matrix()
+            gtsfm_tran = np.array([tx, ty, tz])
+
+            gtsfm_pose = np.eye(4)
+            gtsfm_pose[:3, :3] = gtsfm_rot
+            gtsfm_pose[:3, 3] = gtsfm_tran
+
+            center_pose = np.eye(4)
+            center_pose[:3, 3] = -1 * means
+
+            alignment_pose = np.eye(4)
+            alignment_pose[:3, :3] = align_rot
+
+            swap_x_z = np.array([[0, 0, -1, 0], [0, -1, 0, 0], [1, 0, 0, 0], [0, 0, 0, 1]])
+
+            final_pose = swap_x_z @ alignment_pose @ center_pose @ gtsfm_pose
```

Comment:
i'd prefer to keep the IO as pure IO, and just have `export_as_colmap_txt()` generate a new GtsfmData instance, and then make the pure 3 io calls